### PR TITLE
Imported Dave's SIMD Vector code

### PR DESCRIPTION
**Pull request by esiragusa**

### CODE REVIEW

**Comment by esiragusa** [comment]
Warnings to be fixed (see http://cdash.seqan.de/viewBuildError.php?type=1&buildid=123247)

File: include/seqan/basic/basic_simd_vector.h [hunk 1]
```diff
@@ -94,25 +94,44 @@ assignValue(TSimdVector &vector, TPosition pos, TValue2 value)
...
+    template <> struct Value<TSimdVector const>:  public Value<TSimdVector> {};                             \
+    template <> struct LENGTH<TSimdVector const>: public LENGTH<TSimdVector> {};                            \
+    SEQAN_DEFINE_SIMD_VECTOR_GETVALUE_(TSimdVector const)                                                   \
+    SEQAN_DEFINE_SIMD_VECTOR_VALUE_(TSimdVector)                                                            \
+    SEQAN_DEFINE_SIMD_VECTOR_VALUE_(TSimdVector const)                                                      \
+    SEQAN_DEFINE_SIMD_VECTOR_ASSIGNVALUE_(TSimdVector)                                                      \
+    template <>                                                                                             \
+    SEQAN_CONCEPT_IMPL((TSimdVector),       (SimdVectorConcept));                                           \
+    template <>                                                                                             \
+    SEQAN_CONCEPT_IMPL((TSimdVector const), (SimdVectorConcept));
```
Comment (weese) [inline include/seqan/basic/basic_simd_vector.h]: @esiragusa You can remove the last ```;```. That should fix the warnings.

Comment (esiragusa) [inline include/seqan/basic/basic_simd_vector.h]: Yep this was a reminder for myself :)

**Comment by esiragusa** [comment]
@weese this PR introduced compile errors on platforms which are not under CI - see e.g. http://cdash.seqan.de/viewBuildError.php?buildid=123258 or http://cdash.seqan.de/viewBuildError.php?buildid=123260 - do you know what could it be?

**Pull request merged**